rtbkit/common/plugin_table.h

```diff
@@ -0,0 +1,109 @@
+/* plugin_table.h
+   Flavio Moreira, 15 December 2014
+   Copyright (c) 2014 Datacratic.  All rights reserved.
+*/
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <boost/any.hpp>
+#include "jml/arch/spinlock.h"
+
+namespace RTBKIT {
+
+struct PluginTable
```

Comment:
Ok so now you have one static instance of the PluginTable per Factory type ?
I think I'm ok with that, @EricRobert ?